clients/da-vinci-client/src/main/java/com/linkedin/davinci/storage/StorageService.java

```diff
@@ -205,10 +220,15 @@ private void restoreAllStores(
             storageEngine = openStore(storeConfig, () -> null);
           } catch (Exception e) {
             if (ExceptionUtils.recursiveClassEquals(e, RocksDBException.class)) {
-              LOGGER.error("Could not load the following store : " + storeName, e);
-              aggVersionedStorageEngineStats.recordRocksDBOpenFailure(storeName);
+              LOGGER.warn("Encountered RocksDB error while opening store: {}", storeName, e);
+              if (deleteStorageEngine(storeName)) {
+                LOGGER.info("Store {} does not exist, will delete it.", storeName);
+                factory.removeStorageEngine(storeName);
+              }
               continue;
             }
+            LOGGER.error("Could not load the following store : " + storeName, e);
```

Comment:
Maybe we should consolidate this logic into the above function to make it more testable.

Such as renaming the above method to be: `cleanupStorageEngineWhenEncounteringError` and that static function can take three params:
```
Exception
readOnlyStoreRepository
storageEngineName
```
So you can test almost all the new logic.

Reply:
good idea!  Let me add some unit test